tt_metal/hw/ckernels/grayskull/metal/llk_api/llk_sfpu/ckernel_sfpu_mask.h

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+#include "noc_nonblocking_api.h"
+
+#include "sfpi.h"
+
+using namespace sfpi;
+
+namespace ckernel {
+namespace sfpu {
+
+
+template <bool APPROXIMATION_MODE, int ITERATIONS=4>
+inline void calculate_mask()
+{
+    bool exponent_size_8 = true;
+    for (int d = 0; d < ITERATIONS; d++)
```

Comment:
Can we unroll? Or do we know if the compiler will auto-unroll for us?

Reply:
we definitely do not know, and whether or not unrolling is a win is a bit hit or miss, but generally it is